hack/boilerplate/boilerplate.py

```diff
@@ -70,25 +102,52 @@ def file_passes(filename, extension, ref, regexs):
 
     return True
 
-def main():
-    if len(sys.argv) < 3:
-        PrintError("usage: %s extension FILENAME [FILENAMES]" % sys.argv[0])
-        return False
-
-    basedir = os.path.dirname(os.path.abspath(__file__))
-
-    extension = sys.argv[1]
-    # argv[0] is the binary, argv[1] is the extension (go, sh, py, whatever)
-    filenames = sys.argv[2:]
-
-    ref_filename = basedir + "/boilerplate." + extension + ".txt"
-    try:
-        ref_file = open(ref_filename, 'r')
-    except:
-        # No boilerplate template is success
-        return True
-    ref = ref_file.read().splitlines()
-
+def valid_file_extension(filename, extensions):
+    for ext in extensions:
+        if filename.endswith("." + ext):
+            return True
+    return False
+
+def normalize_files(files):
```

Comment:
`os.walk()` is only used if this is called with no arguments. We can cut back the walk code and not ignore godeps/third_path/output, but that would just mean we walk more of the filesystem and have a bigger file list which gets stripped down in `normalize_files`. So the stuff in the walk code is totally a speed optimization.

We need to do the pathname stripping in `normalize_files` in case we are called with command line args (from pre-commit hooks) which list files in Godeps or third_party and we need to ignore them.  (actually I don't think we need _output in there but it's not hurting anything.)